packages/subapp-server/lib/fastify-plugin.js

```diff
@@ -56,8 +56,15 @@ function makeRouteHandler({ path, routeRenderer, routeOptions }) {
       const data = context.result;
       const status = data.status;
 
-      const cspHeader = getCSPHeader({ styleNonce, scriptNonce });
-
+      let cspHeader;
+      /** If csp headers are provided by application in route options then use that otherwise generate CSP headers */
+      if(routeOptions.getCSPHeader && typeof routeOptions.getCSPHeader === "function"){
+        const rawCSPHeader = routeOptions.getCSPHeader({ styleNonce, scriptNonce });
```

Comment:
Since this is to set CSP header, I guess a better suitable function name would be something like `setCSPHeader` instead of `getCSP*` in routeOptions. (Corresponding change in route as well)

Reply:
Personally I prefer getCSPHeader or may be just `cspHeaderValues`, since the user is not really setting the header, they will be only returning the list of header values (similar to getCSPHeader function in our package).

Reply:
@arunvishnun @RahulTinku  what are your thoughts on  `cspHeaderValues` as the option name?

Reply:
Agree, or probably `getCSPDirectives`.

Reply:
I agree with `cspHeaderValues` as we are just adding the values. Setting the headers would be taken care by subapp-server.